fakenewsredis.py

```diff
@@ -701,6 +704,55 @@ def sort(self, name, start=None, num=None, by=None, get=None, desc=False,
         except KeyError:
             return []
 
+    def eval(self, script, numkeys, *keys_and_args):
+        """
+        Execute the Lua ``script``, specifying the ``numkeys`` the script
+        will touch and the key names and argument values in ``keys_and_args``.
+        Returns the result of the script.
+        In practice, use the object returned by ``register_script``. This
+        function exists purely for Redis API completion.
+        """
+        lua_runtime = LuaRuntime(unpack_returned_tuples=True)
+
+        raw_lua = """
+        function(KEYS, ARGV, callback)
+            redis = {{}}
+            redis.call = callback
+            {body}
+        end
+        """.format(body=script)
+        keys = (None,) + keys_and_args[:numkeys]
+        args = (None,) + keys_and_args[numkeys:]
+
+        lua_func = lua_runtime.eval(raw_lua)
+        result = lua_func(
+            keys,
+            args,
+            self._lua_callback
+        )
+        if lua_type(result) == 'table':
+            # Convert Lua tables into lists, starting from index 1, mimicking the behavior of StrictRedis.
+            result_list = []
+            for index in count(1):
+                if index not in result:
+                    break
+                item = result[index]
+                result_list.append(
+                    item.encode() if isinstance(item, str) and not isinstance(item, bytes) else item
+                )
+            return result_list
+        return result
+
+    def _lua_callback(self, op, *args):
+        special_cases = {
+            'del': self.delete,
+            'decrby': self.decr,
+            'incrby': self.incr
+        }
+        op = op.lower()
+        func = special_cases[op] if op in special_cases else getattr(self, op)
```

Comment:
There are presumably attributes of `self` that shouldn't be visible to Lua - for a start, anything with an underscore prefix, plus things like `delete` and `from_url`. Also, when `self` is a FakeRedis rather than a FakeStrictRedis, it presumably should use the overload from the FakeStrictRedis base class.

Reply:
Probably yes. This is a bit of a hack, but I figure the tradeoff is that it doesn't have a huge mapping from Redis commands to Python functions that needs to be maintained as more Redis functions get added. I'll make it a bit more selective.